### ISSUE TITLE
After update to Zammad 6.0.0 brand logo is broken (only docker version)

### ISSUE THREAD

**Issue by mgruner** [zammad/zammad-docker-compose#354]
Please see: https://community.zammad.org/t/after-update-to-zammad-6-0-0-brand-logo-is-broken-only-docker-version/11654

The issue is probably that the nginx container now only has the default /opt/zammad file system, and never runs the Rails initializer that writes the logo file to `assets/` for this container.

If that's true, there are two options:

- Change the docker-compose to also share the relevant folder, until 6.1 is released
- Back port our [logo handling refactoring](https://github.com/zammad/zammad/commit/afdb85baf85909ef2dd611f241b4625f7acff91c), which would switch it to a different URL handled by the application instead of nginx. We don't usually make such changes in a stable branch, we'd need to clarify that.

**Comment by mgruner** [comment]
@monotek is it feasible to restore the old behaviour for the assets folder for 6.0?

**Comment by monotek** [comment]
Yes, thats probably the problem.
I guess only the initcontainer will have the asset right now.
I ´ll think about a fix but i guess i will have not time until next week.

**Comment by mgruner** [comment]
Thanks!

I guess that every container which runs the Rails stack will have the asset, as the initializer is executed which writes the file. Maybe copying the files to the nginx container would be enough.

**Comment by monotek** [comment]
Can we somehow store the image in /var?

**Comment by mgruner** [comment]
No, nginx will not be able to find it there.

But what about copying `assets/images/` from the init container to the nginx container, same location? Since the init container creates the file, this might work. The other containers should also create the file during Rails startup.

**Comment by monotek** [comment]
Not sure how that should be done.
There is not way to copy data from one container to another, from inside a container.
At least not out of the box, with our containers. Woudl likely also need elevated permissions. 
Maybe using /var as shared storage but that would be really ugly.

**Comment by monotek** [comment]
Even using a new volume would need to modify the entrypoint again :/
Maybe creating zammad 6.1 and adding the backport you've talked about would be the least effort.

**Comment by monotek** [comment]
related: https://github.com/zammad/zammad-helm/issues/191

**Comment by waja** [comment]
How important is the (missing) brand logo to invest so much resources to fix this, when a fix is available in the next Zammad version, which is caring about this?

**Comment by sniper7kills** [comment]
I would argue that a missing brand logo isn't important on its own, but due to the fact it is breaking the startup in Kubernetes (possibly docker-compose?) due to the init script failing its more important than just having a missing logo.

**Comment by monotek** [comment]
A temporary workaround could be to disable the readonly filesystem setting for the container.

**Comment by mgruner** [comment]
Another idea: we could change the docker entrypoint script so that `zammad-nginx` would call a Rails command like `rails r true` before starting nginx. This should cause the logo file to be written for the nginx container. We only need to do that for `stable`, not `develop`.

Regarding Helm I guess we need to switch back to `rw` file system for 6.0, and can try changing this back for 6.1.

What do you think? I can try the proposed entrypoint change early next week.

**Comment by waja** [comment]
> I would argue that a missing brand logo isn't important on its own, but due to the fact it is breaking the startup in Kubernetes (possibly docker-compose?) due to the init script failing its more important than just having a missing logo.

Until now I've not recognised any issue about this on my compose setups.

**Comment by monotek** [comment]
I guess it only happens with custom / autodetected logo?

**Comment by mgruner** [comment]
Yes, that's correct. I have a fix for the entrypoint script which solves this problem. Will open this issue on `zammad/zammad` and a public MR for feedback.

**Comment by mgruner** [comment]
Please see PR https://github.com/zammad/zammad/pull/4684

**Comment by monotek** [comment]
Imho the pr can be merged and the new image version used here...

**Comment by mgruner** [comment]
This should be solved by https://github.com/zammad/zammad/commit/63a731c05d1381691a8eec8c7b3cef7ad313a992.

@monotek for the Helm chart I guess we could switch back to rw filesystem for now, and try ro again for Zammad 6.1? The fix above is not needed for `develop`, I confirmed that logo handling works correctly without FS writes at all.
